packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -938,6 +961,12 @@ export function getValidatorApi(
         // Requesting duties for next epoch is allow since they can be predicted with high probabilities.
         // @see `epochCtx.getBeaconProposersNextEpoch` JSDocs for rationale.
         indexes = state.epochCtx.getBeaconProposersNextEpoch();
+      } else if (epoch === stateEpoch - 1) {
+        const indexesPrevEpoch = state.epochCtx.getBeaconProposersPrevEpoch();
+        if (indexesPrevEpoch === null) {
+          throw new ApiError(500, `Proposer duties for previous epoch ${epoch} not yet initialized`);
+        }
+        indexes = indexesPrevEpoch;
       } else {
```

Comment:
may want to replace this multiple `if/else` clauses by `switch/case`

Reply:
I avoided this this to reduce the diff but since initial review is done, refactored it now